### PR TITLE
Publish Docker image to GHCR

### PR DESCRIPTION
**Pull request by agoelzer**
this also includes a fix where currently the git SHA doesn't show up in the build number in the UI

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -5,6 +5,47 @@ on:
...
+        run: |
+          make build-image
+      - name: Setup Integration Test Environment
+        run: |
+          make setup-integration-tests
+      - name: Run Integration Tests
+        run: |
+          make run-integration-tests-only
+      - name: Teardown Integration Test Environment
+        if: success() || failure()
```
Comment (adriansuarez) [inline .github/workflows/build.yaml:29]: When would this step not run if it runs on success or failure?

Comment (adriansuarez) [reply .github/workflows/build.yaml:29]: According to [this](https://stackoverflow.com/a/58859404), the `success() || failure()` idiom is to not run on cancelled.

Why not change to `if: '!cancelled()'`, which is clearer?

**Review by adriansuarez — APPROVED** [review]
Looks good.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
         run: |
           make run-integration-tests-only
       - name: Teardown Integration Test Environment
-        if: success() || failure()
+        if: !cancelled()
```
Comment (adriansuarez) [inline .github/workflows/build.yaml:29]: It needs to be quoted, i.e. `if: '!cancelled()'`, because I think `!` is a special character in YAML.

Comment (agoelzer) [reply .github/workflows/build.yaml]: Done.

**Pull request merged**